plugins/srp.c

```diff
@@ -1445,14 +1445,15 @@ static int LayerInit(srp_options_t *opts, context_t *text,
 	oparams->mech_ssf = opt->ssf;
 
 	/* Initialize the HMACs */
-	text->hmac_md = EVP_get_digestbyname(opt->evp_name);
-        text->hmac_send_ctx = HMAC_CTX_new();
-	HMAC_Init_ex(text->hmac_send_ctx, text->K, text->Klen, text->hmac_md, NULL);
-        text->hmac_recv_ctx = HMAC_CTX_new();
-	HMAC_Init_ex(text->hmac_recv_ctx, text->K, text->Klen, text->hmac_md, NULL);
+	params[0] = OSSL_PARAM_construct_utf8_string("digest", opt->evp_name, 0);
+	params[1] = OSSL_PARAM_construct_end();
+	text->hmac_send_ctx = EVP_MAC_CTX_new(EVP_MAC_fetch(NULL, "HMAC", NULL));
```

Comment:
Would prefer the EVP_MAC_fetch() was only called once, and the return value reused for both CTX_new() calls.

Reply:
Applied.